examples/basics/src/components/Welcome.astro

```diff
@@ -0,0 +1,200 @@
+---
+import astroLogo from '../assets/astro.svg';
+---
+
+<div id="container">
+	<main>
+		<section id="hero">
+			<a href="https://astro.build"
+				><img src={astroLogo.src} width="115" height="48" alt="Astro Logo" /></a
+			>
+			<h1>
+				To get started, open the <button
+					onclick=`fetch("/__open-in-editor?file=${process.cwd()}/src/pages/index.astro");`
+					title="Click to open in your editor"><code><pre>src/pages</pre></code></button
+				> directory in your project.
+			</h1>
```

Comment:
I am not sure how much this applies to other screen readers besides NVDA (would appreciate it if anyone could check in JAWS/VoiceOver), but since the button's content is wrapped in `<code><pre>src/pages</pre><code>`, the screen reader is unable to read it as a sentence.

Instead of getting:
"To get started, open the *button* src/pages directory in your project."

You actually get:
"To get started, open the" (you have to ask the screen reader to read the next line/content) "button src/pages" (ask yet again) "directory in your project."

This is not a nice experience, removing the `<code>` and `<pre>`  elements solves the issue but would need changing some of the CSS for it.

The best solution for the least effort I could find is to:

Add the following CSS:
```css
.sr-only {
	position: absolute;
	width: 1px;
	height: 1px;
	padding: 0;
	margin: -1px;
	overflow: hidden;
	clip: rect(0, 0, 0, 0);
	white-space: nowrap;
	border-width: 0;
}
```

and then:
```suggestion
			<h1>
				To get started, open the <button
					onclick=`fetch("/__open-in-editor?file=${process.cwd()}/src/pages/index.astro");`
					title="Click to open in your editor"><span class="sr-only">src/pages</span><code aria-hidden="true"><pre>src/pages</pre></code></button
				> directory in your project.
			</h1>
```

This will add a direct children "src/pages" span content to the button that is available in the accessibility tree but invisible. While, at the same time, makes `<code>` hidden from the accessibility tree but visible.

Reply:
I’d additionally recommend avoiding the `title` attribute in favour of `sr-only` text. Attributes like `title` and `aria-label` are generally not as accessible as good old text. For example, they usually don’t respond to browser translation tools.

(And while `title` _seems_ nice because it can have a tooltip, we should never depend on that tooltip as there are many many cases where that visible display will never be triggered like touch screens or keyboard focus.)